packages/neutron-sdk/src/interchain_queries/types.rs

```diff
@@ -186,6 +211,204 @@ impl KVReconstruct for Balances {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+/// A structure that can be reconstructed from **StorageValues**'s for the **Bank Total Interchain Query**.
+/// Contains total supply for specific denom that are held on remote chain.
+pub struct TotalSupply {
+    pub coins: Vec<Coin>,
+}
+
+impl KVReconstruct for TotalSupply {
+    fn reconstruct(storage_values: &[StorageValue]) -> NeutronResult<TotalSupply> {
+        let mut coins: Vec<Coin> = vec![];
+
+        for kv in storage_values {
+            let denom = get_total_supply_denom(kv.key.to_base64());
+            let amount = get_total_supply_amount(kv.value.to_base64());
```

Comment:
You are encoding values here to base64 and decoding it back to bytes inside `get_total_supply_denom` which is inefficient as hell.
And I don't understand why, because you can directly call `get_denom_value` with passing bytes value of `kv.key` and `kv.value` variables.

Please, fix that

Reply:
The same about `get_total_supply_amount()`